codecov_cli/services/upload/coverage_file_finder.py

```diff
@@ -163,26 +166,76 @@
+        self.disable_search = disable_search
 
     def find_coverage_files(self) -> typing.List[UploadCollectionResultFile]:
-        regex_patterns_to_include = globs_to_regex(coverage_files_patterns)
         regex_patterns_to_exclude = globs_to_regex(coverage_files_excluded_patterns)
+        coverage_files_paths = []
+        user_coverage_files_paths = []
+        if self.explicitly_listed_files:
+            user_filenames_to_include = []
+            files_excluded_but_user_includes = []
+            for file in self.explicitly_listed_files:
+                user_filenames_to_include.append(file.name)
+                if regex_patterns_to_exclude.match(file.name):
+                    files_excluded_but_user_includes.append(str(file))
+            if files_excluded_but_user_includes:
+                logger.warning(
+                    "Some files being explicitly added are found in the list of excluded files for upload.",
+                    extra=dict(
+                        extra_log_attributes=dict(
+                            files=files_excluded_but_user_includes
+                        )
+                    ),
+                )
+            regex_patterns_to_include = globs_to_regex(user_filenames_to_include)
+            multipart_include_regex = globs_to_regex(
+                [str(path.resolve()) for path in self.explicitly_listed_files]
+            )
+            user_coverage_files_paths = search_files(
+                self.project_root,
+                default_folders_to_ignore + self.folders_to_ignore,
+                filename_include_regex=regex_patterns_to_include,
+                filename_exclude_regex=regex_patterns_to_exclude,
+                multipart_include_regex=multipart_include_regex,
+            )
+        if not self.disable_search:
+            regex_patterns_to_include = globs_to_regex(coverage_files_patterns)
+            coverage_files_paths = search_files(
+                self.project_root,
+                default_folders_to_ignore + self.folders_to_ignore,
+                filename_include_regex=regex_patterns_to_include,
+                filename_exclude_regex=regex_patterns_to_exclude,
+            )
+        result_files = [
+            UploadCollectionResultFile(path)
+            for path in coverage_files_paths
+            if coverage_files_paths
+        ]
+        user_result_files = [
+            UploadCollectionResultFile(path)
+            for path in user_coverage_files_paths
+            if user_coverage_files_paths
+        ]
 
-        coverage_files_paths = search_files(
-            self.project_root,
-            default_folders_to_ignore + self.folders_to_ignore,
-            filename_include_regex=regex_patterns_to_include,
-            filename_exclude_regex=regex_patterns_to_exclude,
-        )
-
-        return [UploadCollectionResultFile(path) for path in coverage_files_paths]
+        return result_files + user_result_files
```

Comment:
You should consider using sets instead of lists to avoid having duplicated paths returned (for example if a user lists a file that is found in the search it would be present in both lists and returned twice)